sqlite/ext/comdb2/threadpools.c

```diff
@@ -0,0 +1,288 @@
+}
+
+static int systblThreadPoolsColumn(sqlite3_vtab_cursor *cur,
+                                   sqlite3_context *ctx, int pos)
+{
+    /* We already have a lock on threadpools. */
+    systbl_threadpools_cursor *pCur = (systbl_threadpools_cursor *)cur;
+    struct thdpool *pool = pCur->pool;
+
+    /* Lock the pool. */
+    thdpool_lock(pool);
+
+    switch (pos) {
+    case COLUMN_NAME:
+        sqlite3_result_text(ctx, thdpool_get_name(pool), -1, NULL);
+        break;
+    case COLUMN_STATUS:
+        sqlite3_result_text(
+            ctx, thdpool_get_status(pool) ? "stopped" : "running", -1, NULL);
+        break;
+    case COLUMN_NUM_THD:
+        sqlite3_result_int(ctx, thdpool_get_nthds(pool));
+        break;
+    case COLUMN_FREE_THD:
+        sqlite3_result_int(ctx, thdpool_get_nfreethds(pool));
+        break;
+    case COLUMN_PEAK_THD:
+        sqlite3_result_int(ctx, thdpool_get_peaknthds(pool));
+        break;
+    case COLUMN_NUM_CREATES:
+        sqlite3_result_int(ctx, thdpool_get_creates(pool));
+        break;
+    case COLUMN_NUM_EXITS:
+        sqlite3_result_int(ctx, thdpool_get_exits(pool));
+        break;
+    case COLUMN_NUM_PASSED:
+        sqlite3_result_int(ctx, thdpool_get_passed(pool));
+        break;
+    case COLUMN_NUM_ENQUEUED:
+        sqlite3_result_int(ctx, thdpool_get_enqueued(pool));
+        break;
+    case COLUMN_NUM_DEQUEUED:
+        sqlite3_result_int(ctx, thdpool_get_dequeued(pool));
+        break;
+    case COLUMN_NUM_TIMEOUT:
+        sqlite3_result_int(ctx, thdpool_get_timeouts(pool));
+        break;
+    case COLUMN_NUM_FAILED_DISPATCHES:
+        sqlite3_result_int(ctx, thdpool_get_failed_dispatches(pool));
+        break;
+    case COLUMN_MIN_THDS:
+        sqlite3_result_int(ctx, thdpool_get_minnthd(pool));
+        break;
+    case COLUMN_MAX_THDS:
+        sqlite3_result_int(ctx, thdpool_get_maxnthd(pool));
+        break;
+    case COLUMN_PEAK_QUEUE:
+        sqlite3_result_int(ctx, thdpool_get_peakqueue(pool));
+        break;
+    case COLUMN_MAX_QUEUE:
+        sqlite3_result_int(ctx, thdpool_get_maxqueue(pool));
+        break;
+    case COLUMN_QUEUE:
+        sqlite3_result_int(ctx, thdpool_get_nqueuedworks(pool));
+        break;
+    case COLUMN_LONG_WAIT_MS:
+        sqlite3_result_int(ctx, thdpool_get_longwaitms(pool));
+        break;
+    case COLUMN_LINGER_SECS:
+        sqlite3_result_int(ctx, thdpool_get_lingersecs(pool));
+        break;
+    case COLUMN_STACK_SIZE:
+        sqlite3_result_int(ctx, thdpool_get_stacksz(pool));
+        break;
+
+    case COLUMN_MAX_QUEUE_OVERRIDE:
+        sqlite3_result_int(ctx, thdpool_get_maxqueueoverride(pool));
+        break;
+    case COLUMN_MAX_QUEUE_AGE_MS:
+        sqlite3_result_int(ctx, thdpool_get_maxqueueagems(pool));
+        break;
+    case COLUMN_EXIT_ON_CREATE_FAIL:
+        sqlite3_result_text(
+            ctx, YESNO(thdpool_get_exit_on_create_fail(pool) == 1), -1, NULL);
```

Comment:
Zero for false and non-zero for true. So you don't need to compare the value with 1.